pageserver/src/walredo.rs

```diff
@@ -689,22 +693,59 @@ impl PostgresRedoProcess {
         let stdout = child.stdout.take().unwrap();
         let stderr = child.stderr.take().unwrap();
 
-        set_nonblock(stdin.as_raw_fd())?;
-        set_nonblock(stdout.as_raw_fd())?;
-        set_nonblock(stderr.as_raw_fd())?;
+        macro_rules! set_nonblock_or_log_err {
+            ($file:ident) => {{
+                let res = set_nonblock($file.as_raw_fd());
+                if let Err(e) = &res {
+                    error!(error = %e, concat!("set_nonblock failed on ", stringify!($file)));
+                }
+                res
+            }};
+        }
+        set_nonblock_or_log_err!(stdin)?;
+        set_nonblock_or_log_err!(stdout)?;
+        set_nonblock_or_log_err!(stderr)?;
```

Comment:
no need to change on this PR but would much rather have the logging happen only at the site of handling the error.

one could do this instead for the same effect:

```suggestion
        macro_rules! set_nonblock_with_target {
            ($file:ident) => {{
                set_nonblock($file.as_raw_fd())
                  .context(concat!("set_nonblock failed on ", stringify!($file)))
            }};
        }
        }
        set_nonblock_with_target!(stdin)?;
        set_nonblock_with_target!(stdout)?;
        set_nonblock_with_target!(stderr)?;
```

But as we discussed, most of the comments root cause was the panic!=abort confusion. This could even just be left as it was.

Reply:
I generally agree, but then we lose the tracing span context, don't we?
I.e., 
```
 #[instrument(skip_all,fields(tenant_id=%tenant_id, pg_version=pg_version))]
```

That being said, I guess we should also be logging the PID. Will push another fixup.

Reply:
Oki, good point about the additional context. Maybe it is better here, in case we see that many defunct processes end up being created because of this handling...